core/src/banking_stage/latest_unprocessed_votes.rs

```diff
@@ -265,21 +265,18 @@ impl LatestUnprocessedVotes {
         let with_latest_vote = |latest_vote: &RwLock<LatestValidatorVotePacket>,
                                 vote: LatestValidatorVotePacket|
          -> Option<LatestValidatorVotePacket> {
-            let should_try_update = Self::allow_update(
+            if Self::allow_update(
                 &vote,
                 &latest_vote.read().unwrap(),
                 should_replenish_taken_votes,
-            );
-            if should_try_update {
+            ) {
                 let mut latest_vote = latest_vote.write().unwrap();
-                if Self::allow_update(&vote, &latest_vote, should_replenish_taken_votes) {
```

Comment:
To clarify, the second `allow_update` was checking against a potentially updated `latest_vote` (by other threads). Since it is single thread now, the second `latest_vote` at ln273 shall be the same as the one outside, hence the check is redundant. Is it correct?

Reply:
Exactly - the first `allow_update` was checked with read permissions. If it could be updated it grabbed write-lock and checked the condition again.

In a single-thread this makes zero sense to keep